keps/sig-storage/3476-volume-group/README.md

```diff
@@ -0,0 +1,1680 @@
+}
+
+message CreateSnapshotResponse {
+  Snapshot snapshot = 1;
+  …
+  string group_snapshot_id = 2 [(alpha_field) = true];
+}
+```
+
+#### DeleteVolumeGroupSnapshot
+
+```
+message DeleteVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // The ID of the group snapshot to be deprovisioned.
+  // This field is REQUIRED.
+  string group_snapshot_id = 1;
+
+  // Secrets required by plugin to complete group snapshot deletion request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 2 [(csi_secret) = true];
+}
+
+message DeleteVolumeGroupSnapshotResponse {
+  // Intentionally empty.
+}
+```
+
+#### ControllerGetVolumeGroupSnapshot
+
+```
+message ControllerGetVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // The ID of the group snapshot to fetch current group snapshot information for.
+  // This field is REQUIRED.
+  string group_snapshot_id = 1;
+}
+
+message ControllerGetVolumeGroupSnapshotResponse {
+  option (alpha_message) = true;
+
+  // This field is REQUIRED
+  VolumeGroupSnapshot group_snapshot = 1;
+}
+```
+
+#### ListVolumeGroupSnapshots
+
+```
+message ListVolumeGroupSnapshotsRequest {
+  option (alpha_message) = true;
+
+  // If specified (non-zero value), the Plugin MUST NOT return more
+  // entries than this number in the response. If the actual number of
+  // entries is more than this number, the Plugin MUST set `next_token`
+  // in the response which can be used to get the next page of entries
+  // in the subsequent `ListVolumeGroupSnapshots` call. This field is OPTIONAL. If
+  // not specified (zero value), it means there is no restriction on the
+  // number of entries that can be returned.
+  // The value of this field MUST NOT be negative.
+  int32 max_entries = 1;
+
+  // A token to specify where to start paginating. Set this field to
+  // `next_token` returned by a previous `ListVolumeGroupSnapshots` call to get the
+  // next page of entries. This field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string starting_token = 2;
+}
+
+message ListVolumeGroupSnapshotsResponse {
+  option (alpha_message) = true;
+
+  message Entry {
+    // This field is REQUIRED
+    VolumeGroupSnapshot group_snapshot = 1;
+  }
+
+  repeated Entry entries = 1;
+
+  // This token allows you to get the next page of entries for
+  // `ListVolumeGroupSnapshots` request. If the number of entries is larger than
+  // `max_entries`, use the `next_token` as a value for the
+  // `starting_token` field in the next `ListVolumeGroupSnapshots` request. This
+  // field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string next_token = 2;
+}
+```
+
+### Alternatives
+
+#### Immutable VolumeGroup
+
+During the design discussions, an immutable VolumeGroup was considered but was removed because this would add lots of complexity to the design without much gain.
```

Comment:
I think it would also make it impossible to support the current way PVCs are provisioned in a statefulset.